src/Pipes/GZip.hs

```diff
@@ -62,8 +62,8 @@ decompress' = Pipes.Zlib.decompress' gzWindowBits
 compress
   :: MonadIO m
   => Pipes.Zlib.CompressionLevel
-  -> Proxy x' x () B.ByteString m r -- ^ Decompressed stream
-  -> Proxy x' x () B.ByteString m r -- ^ Compressed stream
+  -> Producer B.ByteString m r -- ^ Decompressed stream
+  -> Producer' B.ByteString m r -- ^ Compressed stream
```

Comment:
To support my claim that the new signature is actually _more_ general, consider the case where we pass in a `Producer`.  Then a function with the old signature is forced to return a `Producer` - but a function with the new signature can return a `Producer'`.

If you still have doubts: we can ask GHC to confirm.  As I say, the inferred type is `Producer -> Producer`'.  But if we explicitly annotate as `Producer' -> Producer'`, then the code still compiles: it's a special case of the inferred type.  `Producer -> Producer'` really is more general.